packages/core/src/components/forms/numericInput.tsx

```diff
@@ -105,6 +107,12 @@ export interface INumericInputProps extends IIntentProps, IProps {
      */
     leftIcon?: IconName | MaybeElement;
 
+    /**
+     * The locale name, which is passed to the component to format the number and allowing to type the number in the specific locale.
```

Comment:
I think we should link to this documentation of the browser locales API: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl#Locale_identification_and_negotiation